views/user/authorized_index.ejs

```diff
@@ -3663,6 +3663,15 @@
                                      });
                     }
                 }
+				let allDates = [];
+				for (let data of chartData) {
+					if (data && data.assignmentDates) {
+						allDates = allDates.concat(data.assignmentDates);
+					}
+				}
+				allDates = allDates.map(dateStr => new Date(dateStr));
+				const earliestDate = new Date(Math.min(...allDates));
+				const latestDate = new Date(Math.max(...allDates));
```

Comment:
Is there a reason to calculate these instead of using the `startDate` and `endDate` variables that exist in this scope? Also, spaces not tabs.